tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -820,17 +834,93 @@ bool DeclResultIdMapper::createStageVars(
 
     if (asInput) {
       *value = theBuilder.createLoad(typeId, varId);
+
+      // Fix ups for corner cases
+
+      // Special handling of SV_TessFactor DS patch constant input.
+      // TessLevelOuter is always an array of size 4 in SPIR-V, but
+      // SV_TessFactor could be an array of size 2, 3, or 4 in HLSL. Only the
+      // relevant indexes must be loaded.
+      if (semanticKind == hlsl::Semantic::Kind::TessFactor &&
+          hlsl::GetArraySize(type) != 4) {
```

Comment:
Hmm, nice! We should probably use this method for other places too! Not for this pull request.